src/v/cloud_storage/partition_manifest.cc

```diff
@@ -575,16 +575,19 @@ void partition_manifest::disable_permanently() {
     _last_uploaded_compacted_offset = model::offset::max();
 }
 
-void partition_manifest::subtract_from_cloud_log_size(size_t to_subtract) {
+size_t partition_manifest::subtract_from_cloud_log_size(size_t to_subtract) {
     if (to_subtract > _cloud_log_size_bytes) {
         vlog(
           cst_log.warn,
           "Invalid attempt to subtract from cloud log size of {}. Setting it "
           "to 0 to prevent underflow",
           _ntp);
+        auto prev_sz = _cloud_log_size_bytes;
         _cloud_log_size_bytes = 0;
+        return prev_sz;
```

Comment:
i would have rewritten this function as

```cpp
auto removable=std::min(_cloud_log_size_bytes, to_subtract);
if(removable!=to_subtract){vlog(warn, "...");}
_cloud_log_size_bytes-=removable;
return removable;
```

or are you referring to the fact that i should leave this function alone and just use total_replaced_size, when returning from add()

Reply:
It's a bit strange that we're returning either size of one segment or the size of the full log. It's difficult to follow the logic. It'd be more obvious if `total_replaced_size` were used.  I think it's correct to do so because in case of underflow we're likely dealing with some bug and the log size is computed incorrectly. The `total_replaced_size` couldn't be incorrect because we just got this value out of the manifest (which is our source of truth).